modin/pandas/dataframe.py

```diff
@@ -385,19 +389,25 @@ def groupby(
             # fastpath for multi column groupby
             if axis == 0 and all(
                 (
-                    (hashable(o) and (o in self))
+                    is_label(self, o, axis)
                     or isinstance(o, Series)
-                    or (is_list_like(o) and len(o) == len(self.axes[axis]))
+                    or (
+                        is_list_like(o)
+                        and len(o) == len(self._query_compiler.get_axis(axis))
+                    )
```

Comment:
changed `self.axes` to `self._query_compiler.get_axis` to do not trigger unnecessary axis computations for lazy backends.